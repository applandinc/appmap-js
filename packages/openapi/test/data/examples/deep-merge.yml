- class: Hash
  value: '{}'
  properties:
    - name: id
      class: Integer
    - name: created_at
      class: String
    - name: updated_at
      class: String
    - name: uuid
      class: String
    - name: data
      class: Hash
      properties:
        - name: events
          class: Array
          items:
            - class: Hash
        - name: version
          class: String
        - name: classMap
          class: Array
          items:
            - class: Hash
        - name: metadata
          class: Hash
          properties:
            - name: git
              class: Hash
            - name: name
              class: String
            - name: labels
              class: Array
              items:
                - class: String
            - name: feature
              class: String
            - name: language
              class: Hash
    - name: org_id
      class: Integer
    - name: mapset_id
      class: Integer
    - name: link_sharing
      class: FalseClass
    - name: name
      class: String
- class: Hash
  value: '{}'
  properties:
    - name: events
      class: Array
      items:
        - class: Hash
          properties:
            - name: id
              class: Integer
            - name: event
              class: String
            - name: message
              class: Array
              items:
                - class: Hash
            - name: http_server_request
              class: Hash
        - class: Hash
          properties:
            - name: id
              class: Integer
            - name: self
              class: Hash
            - name: event
              class: String
            - name: sql_query
              class: Hash
        - class: Hash
          properties:
            - name: id
              class: Integer
            - name: path
              class: String
            - name: self
              class: Hash
            - name: event
              class: String
            - name: lineno
              class: Integer
            - name: method_id
              class: String
            - name: defined_class
              class: String
        - class: Hash
          properties:
            - name: id
              class: Integer
            - name: path
              class: String
            - name: event
              class: String
            - name: lineno
              class: Integer
            - name: method_id
              class: String
            - name: parent_id
              class: Integer
            - name: defined_class
              class: String
    - name: version
      class: String
    - name: classMap
      class: Array
      items:
        - class: Hash
          properties:
            - name: name
              class: String
            - name: type
              class: String
            - name: children
              class: Array
              items:
                - class: Hash
            - name: location
              class: String
    - name: metadata
      class: Hash
      properties:
        - name: git
          class: Hash
          properties:
            - name: branch
              class: String
            - name: commit
              class: String
            - name: repository
              class: String
        - name: name
          class: String
        - name: labels
          class: Array
          items:
            - class: String
        - name: feature
          class: String
        - name: language
          class: Hash
          properties:
            - name: name
              class: String
