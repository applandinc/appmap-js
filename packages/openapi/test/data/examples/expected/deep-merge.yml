examples:
  - type: object
    properties:
      created_at:
        type: string
      data:
        type: object
        properties:
          classMap:
            type: array
            items:
              type: object
          events:
            type: array
            items:
              type: object
          metadata:
            type: object
            properties:
              feature:
                type: string
              git:
                type: object
              labels:
                type: array
                items:
                  type: string
              language:
                type: object
              name:
                type: string
          version:
            type: string
      id:
        type: integer
      link_sharing:
        type: boolean
      mapset_id:
        type: integer
      name:
        type: string
      org_id:
        type: integer
      updated_at:
        type: string
      uuid:
        type: string

  - type: object
    properties:
      classMap:
        type: array
        items:
          type: object
          properties:
            children:
              type: array
              items:
                type: object
            location:
              type: string
            name:
              type: string
            type:
              type: string
      events:
        type: array
        items:
          type: object
          properties:
            defined_class:
              type: string
            event:
              type: string
            http_server_request:
              type: object
            id:
              type: integer
            lineno:
              type: integer
            message:
              type: array
              items:
                type: object
            method_id:
              type: string
            parent_id:
              type: integer
            path:
              type: string
            self:
              type: object
            sql_query:
              type: object
      metadata:
        type: object
        properties:
          feature:
            type: string
          git:
            type: object
            properties:
              branch:
                type: string
              commit:
                type: string
              repository:
                type: string
          labels:
            type: array
            items:
              type: string
          language:
            type: object
            properties:
              name:
                type: string
          name:
            type: string
      version:
        type: string

mergeResult:
  type: object
  properties:
    classMap:
      type: array
      items:
        type: object
        properties:
          children:
            type: array
            items:
              type: object
          location:
            type: string
          name:
            type: string
          type:
            type: string
    created_at:
      type: string
    data:
      type: object
      properties:
        classMap:
          type: array
          items:
            type: object
        events:
          type: array
          items:
            type: object
        metadata:
          type: object
          properties:
            feature:
              type: string
            git:
              type: object
            labels:
              type: array
              items:
                type: string
            language:
              type: object
            name:
              type: string
        version:
          type: string
    events:
      type: array
      items:
        type: object
        properties:
          defined_class:
            type: string
          event:
            type: string
          http_server_request:
            type: object
          id:
            type: integer
          lineno:
            type: integer
          message:
            type: array
            items:
              type: object
          method_id:
            type: string
          parent_id:
            type: integer
          path:
            type: string
          self:
            type: object
          sql_query:
            type: object
    id:
      type: integer
    link_sharing:
      type: boolean
    mapset_id:
      type: integer
    metadata:
      type: object
      properties:
        feature:
          type: string
        git:
          type: object
          properties:
            branch:
              type: string
            commit:
              type: string
            repository:
              type: string
        labels:
          type: array
          items:
            type: string
        language:
          type: object
          properties:
            name:
              type: string
        name:
          type: string
    name:
      type: string
    org_id:
      type: integer
    updated_at:
      type: string
    uuid:
      type: string
    version:
      type: string
